src/shuffler-codereview.c

```diff
@@ -0,0 +1,1228 @@
+        // flag for the partner it uses the same flag for the other trainer.
+        RedirectedTrainerFlags |= 1 << partnerObjNum;
+    }
+}
+
+void DeclareWildMon(u8 objNum, u8 partnerObjNum) {
+    MirrorMapData();
+    AdjustedObjectsDirty |= (1 << objNum);
+
+    struct WildMon* wildMon = &AdjustedObjects[objNum].wm;
+
+    // first we roll to see if we get a legendary. if we do, get the legendary
+    // pokemon to spawn. otherwise, get the non-legnedary pokemon to spawn.
+    u16 species;
+    if (WILD_LEGEND_ODDS != 0 && (tinymt32_generate_uint32(&masterSeed) % WILD_LEGEND_ODDS) == 0) {
+        species = possibleWildLegends[tinymt32_generate_uint32(&masterSeed) % POSSIBLE_WILD_MON_LEGENDARY];
+    } else {
+        species = possibleWildMon[tinymt32_generate_uint32(&masterSeed) % POSSIBLE_WILD_MON];
+    }
+
+    // wild pokemon are pretty straightforward. we setup some simple data based
+    // on the species id and that's it.
+    u8 level = GetScaledLevel(distance);
+    wildMon->NoItemDefaultMoves[0].iv = 15;
+    wildMon->NoItemDefaultMoves[0].lvl = level;
+    u16 adjustedSpecies = AdjustSpecies(species, level, EVO_STRAT_WILD);
+    wildMon->NoItemDefaultMoves[0].species = adjustedSpecies;
+    AdjustedTemplates[objNum].graphicsId = OBJ_EVENT_GFX_POKEMON_001 + wildMon->NoItemDefaultMoves[0].species - 1;
+    AdjustedTemplates[objNum].flagId = ShuffledFlagNumberByObjectEventId(objNum + 1);
+    wildMon->partnerObjNum = objNum;
+
+    // check to see if we have a partner object and see if that partner object
+    // has already been generated. if it has, we need to combine the two
+    // objects so the player gets the same battle when interacting with either
+    // one.
+    if (partnerObjNum != 0xFF && (AdjustedObjectsDirty & (1 << partnerObjNum))) {
+        struct WildMon* partnerWildMon = &AdjustedObjects[partnerObjNum].wm;
+
+        // set a flag so when the game attempts to read or write the flag for
+        // the partner it uses the same flag for the other wild pokemon.
+        RedirectedTrainerFlags |= 1 << partnerObjNum;
+        AdjustedTemplates[partnerObjNum].flagId = AdjustedTemplates[objNum].flagId;
+        wildMon->partnerObjNum = partnerObjNum;
+        partnerWildMon->partnerObjNum = objNum;
+
+        // make sure both objects reference both wild pokemon in the same order.
+        // this way no matter which one the player interacts with, they'll get
+        // the same encounter.
+        wildMon->NoItemDefaultMoves[1] = partnerWildMon->NoItemDefaultMoves[0];
+        partnerWildMon->NoItemDefaultMoves[0] = wildMon->NoItemDefaultMoves[0];
+        partnerWildMon->NoItemDefaultMoves[1] = wildMon->NoItemDefaultMoves[1];
+    }
+}
+
+u8 ShufflerGetWildMonPartnerObjNum(u8 objNum) {
+    return AdjustedObjects[objNum - 1].wm.partnerObjNum + 1;
+}
+
+void DeclareItem(u16 objNum) {
+    MirrorMapData();
+    AdjustedObjectsDirty |= (1 << objNum);
+    int i = tinymt32_generate_uint32(&masterSeed) % 6;
```

Comment:
i doesn't seem to used, so don't assign it. Just use it in the switch statement.